src/components/DepositPage.tsx

```diff
@@ -71,15 +82,71 @@ const DepositPage = (props: Props): ReactElement => {
     onConfirmTransaction,
     onToggleDepositWrapped,
   } = props
-
+  const { account, chainId, library } = useActiveWeb3React()
+  const { unstake, amountStaked } = useRewardsHelpers(
+    poolData?.name as PoolName,
+  )
   const [currentModal, setCurrentModal] = useState<string | null>(null)
+  const [liquidityGaugeContract, setLiquidityGaugeContract] =
+    useState<LiquidityGaugeV5>()
+  const lpTokenContract = useLPTokenContract(poolData?.name as PoolName)
   const validDepositAmount = transactionData.to.totalAmount.gt(0)
   const shouldDisplayWrappedOption = isMetaPool(poolData?.name)
   const theme = useTheme()
   const isLgDown = useMediaQuery(theme.breakpoints.down("lg"))
+  const { gauges } = useContext(GaugeContext)
+
+  useEffect(() => {
+    if (!library || !account || !chainId || !poolData) return
+    const gaugeAddr = gauges?.[poolData.poolAddress]?.address ?? ""
+    if (gaugeAddr) {
+      const liquidityGaugeContract = getContract(
+        gaugeAddr,
+        LIQUIDITY_GAUGE_V5_ABI,
+        library,
+        account,
+      ) as LiquidityGaugeV5
+      setLiquidityGaugeContract(liquidityGaugeContract)
+    }
+  }, [library, account, chainId, poolData, gauges])
 
-  const onMigrateToGaugeClick = () => {
-    // TODO: Hook up call to migrate to gauge.
+  const onMigrateToGaugeClick = async () => {
+    if (
+      !liquidityGaugeContract ||
+      !chainId ||
+      !account ||
+      !poolData ||
+      !lpTokenContract
+    )
+      return
+    try {
+      void unstake(amountStaked)
+      await checkAndApproveTokenForTrade(
+        lpTokenContract,
+        liquidityGaugeContract.address,
+        account,
+        await lpTokenContract.balanceOf(account),
```

Comment:
I'm guessing we'd have to `await unstake` before `lpTokenContract.balanceOf(account)` increases > 0